pulp_smash/tests/pulp2/rpm/api_v2/test_iso_crud.py

```
@@ -362,3 +364,70 @@ def test_invalid_file_feed(self):
+        * `Pulp #3100 <https://pulp.plan.io/issues/3100>`_
+
+        Do the following:
+
+        1. Create and sync an ISO repository.
+        2. To recreate the scenario of change in units already in Pulp. After
+           the repository is synced, the feed url will be updated, and the
+           repository will be synced again.
+        3. Assert that number of units remain the same, but the content has
+           changed.
+        """
+        cfg = config.get_config()
+        if selectors.bug_is_untestable(2773, cfg.pulp_version):
+            self.skipTest('https://pulp.plan.io/issues/2773')
+        if selectors.bug_is_untestable(3047, cfg.pulp_version):
+            self.skipTest('https://pulp.plan.io/issues/3047')
+        if selectors.bug_is_untestable(3100, cfg.pulp_version):
+            self.skipTest('https://pulp.plan.io/issues/3100')
+        client = api.Client(cfg, api.json_handler)
+        repo = client.post(REPOSITORY_PATH, _gen_iso_repo(FILE2_FEED_URL))
+        self.addCleanup(client.delete, repo['_href'])
+        repo = client.get(repo['_href'], params={'details': True})
+        utils.sync_repo(cfg, repo)
+        units = utils.search_units(cfg, repo)
+        units_before = len(units)
+        iso_details = {
+            unit['metadata']['name']: unit['metadata']['checksum']
+            for unit in units
+        }
+        importer_config = {'importer_config': {'feed': FILE_FEED_URL}}
+        client.put(repo['importers'][0]['_href'], importer_config)
+        repo = client.get(repo['_href'], params={'details': True})
+        utils.sync_repo(cfg, repo)
+        units = utils.search_units(cfg, repo)
+        units_after = len(units)
+
+        # Assert that the number of units has not changed.
+        self.assertEqual(units_before, units_after)
+
+        # Assert that content of units has changed.
+        for unit in units:
+            if unit['metadata']['name'] in iso_details:
```

What if the first search returned units named `1.iso`, `2.iso` and `3.iso`, and the second search returned units named `2.iso`, `3.iso` and `4.iso`? In that case, the `len()` assertion above would pass, and this `if` statement would hide the error. What's the benefit in allowing that?

---

@Ichimonji10, thanks for your review. It was updated per comments.

Basically we assure that number of units has not changed, that all the previous unit names remain in the repo, and their content has changed.